pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -442,23 +434,31 @@ func (b *Botanist) generateCloudConfigExecutionChart() (*chartrenderer.RenderedC
 				}
 			}
 		}
-
-		workers = append(workers, w)
+		config := map[string]interface{}{
+			"bootstrapToken":    kutil.BootstrapTokenFrom(tokenSecret.Data),
+			"configFilePath":    common.CloudConfigFilePath,
+			"kubernetesVersion": b.Shoot.Info.Spec.Kubernetes.Version,
+			"worker":            w,
+		}
+		var err error
+		config, err = b.InjectShootShootImages(config, common.HyperkubeImageName)
+		if err != nil {
+			return nil, err
+		}
+		return b.ChartApplierShoot.Render(filepath.Join(common.ChartPath, "shoot-cloud-config"), releaseName, metav1.NamespaceSystem, config)
 	}
```

Comment:
I played with it locally, works great. I think the only thing that is missing is what @danielfoehrKn already pointed out earlier:

The old `managedresource-shoot-cloud-config-execution` secret remains orphaned in the system for already existing shoots. While it's not harming in any way it's probably still better to simply delete it in order to prevent confusing of operators that are not that familiar with the code. This deletion code can be added after this line and should have a `TODO` comment that it can be removed in a future version of Gardener.

Reply:
Good catch, what do you think about keep the secret name `managedresource-shoot-cloud-config-execution` for the rbac resources? This way we don't have to worry about migration

Reply:
Hm, I'd say it's easier to follow if the name of the secret containing the `shoot-cloud-config-exeuction-rbac` manifests has also `rbac` in its name. Also, if we would overwrite it like this the `cloud-config-<worker-pool-name>` secret in the `kube-system` namespace of the shoot would be deleted (it would be re-applied by the new managed resource that you create for the worker pool, but it would be shortly not present). This probably doesn't harm too much, but summing those two facts up I'd vote for the small migration logic. WDYT?

Reply:
Okay, let's do the migration

Reply:
Addressed in https://github.com/gardener/gardener/pull/2210/commits/c53a9e855e4eb567b9c1372c9d4243d4ee370313